rmf_task/include/rmf_task/Task.hpp

```diff
@@ -84,16 +85,52 @@ class Task::Booking
     ConstPriorityPtr priority_,
     bool automatic_ = false);
 
-  /// The unique id for this booking
+  /// Constructor
+  ///
+  /// \param[in] id
+  ///   The identity of the booking.
+  ///
+  /// \param[in] earliest_start_time
+  ///   The earliest time that the task may begin.
+  ///
+  /// \param[in] priority
+  ///   The priority of the booking.
+  ///
+  /// \param[in] requester
+  ///   The identifier of the entity that requested this task.
+  ///
+  /// \param[in] request_time
+  ///   The time that this task was booked.
+  ///
+  /// \param[in] automatic
+  ///   Whether this booking was automatically generated, default value as
+  ///   false.
+  Booking(
+    std::string id,
+    rmf_traffic::Time earliest_start_time,
+    ConstPriorityPtr priority,
+    const std::string& requester,
```

Comment:
What do you thing about changing this to just l-value strings? We can std::move the value in the impl. I'd expect most downstream use cases to store the requester in a var. 
```suggestion
    std::string requester,
```